utils/s2n_result.h

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#pragma once
+
+#include <s2n.h>
+#include <stdbool.h>
+
+/* A value which indicates the outcome of a function */
+typedef struct _s2n_result s2n_result;
+
+/* used to signal a successful function return */
+extern const s2n_result S2N_RESULT_OK;
+/* used to signal an error while executing a function */
+extern const s2n_result S2N_RESULT_ERROR;
+
+/* returns true when the result is S2N_RESULT_OK */
+__attribute__((warn_unused_result)) bool s2n_result_is_ok(s2n_result result);
```

Comment:
The attribute might need to be behind a macro if not all compilers support it

Reply:
I know clang and gcc both support it. Are there other compilers that s2n currently supports? I can add a condition to remove it, if so.